# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContextHelper.cpp
# Commit: 5ad078db7d7b
# Full Hash: 5ad078db7d7b025ae400922fde028352a430592d
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 21:58:52
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 1. Rename CanvasRenderingContextHelper::GetContext to GetOrCreateContext. r=jgilbert
#   
#   This patch contains no functional changes but sets us up for refactoring
#   in a later part in this series. It also helps silence a bunch of
#   compiler warnings.
# ==============================================================================

diff -r e59ba9934ece -r 5ad078db7d7b dom/canvas/CanvasRenderingContextHelper.cpp
--- a/dom/canvas/CanvasRenderingContextHelper.cpp	Fri Dec 10 02:57:51 2021 +0000
+++ b/dom/canvas/CanvasRenderingContextHelper.cpp	Fri Dec 10 02:57:51 2021 +0000
@@ -160,17 +160,23 @@
   return ret.forget();
 }
 
-already_AddRefed<nsISupports> CanvasRenderingContextHelper::GetContext(
+already_AddRefed<nsISupports> CanvasRenderingContextHelper::GetOrCreateContext(
     JSContext* aCx, const nsAString& aContextId,
     JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv) {
   CanvasContextType contextType;
   if (!CanvasUtils::GetCanvasContextType(aContextId, &contextType))
     return nullptr;
 
+  return GetOrCreateContext(aCx, contextType, aContextOptions, aRv);
+}
+
+already_AddRefed<nsISupports> CanvasRenderingContextHelper::GetOrCreateContext(
+    JSContext* aCx, CanvasContextType aContextType,
+    JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv) {
   if (!mCurrentContext) {
     // This canvas doesn't have a context yet.
     RefPtr<nsICanvasRenderingContextInternal> context;
-    context = CreateContext(contextType);
+    context = CreateContext(aContextType);
     if (!context) {
       return nullptr;
     }
@@ -185,32 +191,32 @@
     }
 
     mCurrentContext = std::move(context);
-    mCurrentContextType = contextType;
+    mCurrentContextType = aContextType;
 
     nsresult rv = UpdateContext(aCx, aContextOptions, aRv);
     if (NS_FAILED(rv)) {
       // See bug 645792 and bug 1215072.
       // We want to throw only if dictionary initialization fails,
       // so only in case aRv has been set to some error value.
-      if (contextType == CanvasContextType::WebGL1)
+      if (aContextType == CanvasContextType::WebGL1) {
         Telemetry::Accumulate(Telemetry::CANVAS_WEBGL_SUCCESS, 0);
-      else if (contextType == CanvasContextType::WebGL2)
+      } else if (aContextType == CanvasContextType::WebGL2) {
         Telemetry::Accumulate(Telemetry::CANVAS_WEBGL2_SUCCESS, 0);
-      else if (contextType == CanvasContextType::WebGPU) {
+      } else if (aContextType == CanvasContextType::WebGPU) {
         // Telemetry::Accumulate(Telemetry::CANVAS_WEBGPU_SUCCESS, 0);
       }
       return nullptr;
     }
-    if (contextType == CanvasContextType::WebGL1)
+    if (aContextType == CanvasContextType::WebGL1) {
       Telemetry::Accumulate(Telemetry::CANVAS_WEBGL_SUCCESS, 1);
-    else if (contextType == CanvasContextType::WebGL2)
+    } else if (aContextType == CanvasContextType::WebGL2) {
       Telemetry::Accumulate(Telemetry::CANVAS_WEBGL2_SUCCESS, 1);
-    else if (contextType == CanvasContextType::WebGPU) {
+    } else if (aContextType == CanvasContextType::WebGPU) {
       // Telemetry::Accumulate(Telemetry::CANVAS_WEBGPU_SUCCESS, 1);
     }
   } else {
     // We already have a context of some type.
-    if (contextType != mCurrentContextType) return nullptr;
+    if (aContextType != mCurrentContextType) return nullptr;
   }
 
   nsCOMPtr<nsICanvasRenderingContextInternal> context = mCurrentContext;